editor/src/document/document_message_handler.rs

```diff
@@ -76,11 +84,19 @@ impl DocumentsMessageHandler {
 		name
 	}
 
-	fn load_document(&mut self, new_document: DocumentMessageHandler, responses: &mut VecDeque<Message>) {
-		let new_id = generate_uuid();
-		self.active_document_id = new_id;
-		self.document_ids.push(new_id);
-		self.documents.insert(new_id, new_document);
+	fn load_document(&mut self, new_document: DocumentMessageHandler, document_id: u64, replace_first_empty: bool, responses: &mut VecDeque<Message>) {
+		// Special case when loading a document on an empty page
+		if replace_first_empty && self.document_ids.len() == 1 {
+			let first_id = self.document_ids[0];
+			let first_document = self.documents.get(&first_id).unwrap();
+			if first_document.serialize_root().len() == 2 {
```

Comment:
This magic number `2` seems fragile. If `serialize_root()` ever changes, this will break and it'll be hard to notice until eventually somebody finds the regression. Could we create a temporary empty document in here, call `.serialize_root().len()` on it, and compare it to `first_document.serialize_root().len()`?